test/api.js

```diff
@@ -17,14 +17,15 @@ const mainKey = ssbKeys.loadOrCreateSync(path.join(dir, 'secret'))
 
 rimraf.sync(dir)
 
-let sbot = SecretStack({ appKey: caps.shs })
-  .use(require('ssb-db2'))
-  .use(require('ssb-bendy-butt'))
-  .use(require('../'))
-  .call(null, {
-    keys: mainKey,
-    path: dir,
-  })
+function createSbot(config = { keys: mainKey, path: dir }) {
+  return SecretStack({ appKey: caps.shs })
+    .use(require('ssb-db2'))
+    .use(require('ssb-bendy-butt'))
+    .use(require('../'))
+    .call(null, config)
+}
```

Comment:
why are we not using `scuttle-testbot`. It hurts me seeing this code and rimraf and key creation repeated so much.
If there's a good reason it's fine, but I tend to like to do something like this, which has a great simple API, and means one place to maintain test server

```js
// testbot.js
const Server = require('scuttle-testbot')

module.exports = function (opts = {}) {
  // opts = {
  //   startUnclean: Boolean,    - whether to rimraf
  //   keys: SecretKeys
  // }

  const stack = Server
    .use(require('ssb-backlinks'))
    .use(require('ssb-query'))
    .use(require('../'))

  return stack({
    db2: true,
    ...opts
  })
}
```

Reply:
I guess it's because scuttle-testbot wouldn't vastly reduce lines of code for us. Honestly it's just about the rimraf, which is not too bad. The key creation part is something we want to control. scuttle-testbot generates random keys and sometimes we want to control the seed so that the key is always deterministic for the tests.

Reply:
you can create your own keys IF required with scuttle-testbot. (generally not needed apart from persistence/ restart tests which a rare).
the module also takes care of making sure your `path` never collides (unless on the other hand you want it to, in which case you set it. Again rare)

Reply:
MEH